x-pack/plugins/elastic_assistant/server/routes/evaluate/post_evaluate.ts

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { IRouter, Logger } from '@kbn/core/server';
+import { transformError } from '@kbn/securitysolution-es-utils';
+
+import { buildResponse } from '../../lib/build_response';
+import { buildRouteValidation } from '../../schemas/common';
+
+import { ElasticAssistantRequestHandlerContext } from '../../types';
+import { EVALUATE } from '../../../common/constants';
+import { PostEvaluateBody, PostEvaluatePathQuery } from '../../schemas/evaluate/post_evaluate';
+import { performEvaluation } from '../../lib/model_evaluator/evaluation';
+import { callAgentExecutor } from '../../lib/langchain/execute_custom_llm_chain';
+import { callOpenAIFunctionsExecutor } from '../../lib/langchain/executors/openai_functions_executor';
+import { AgentExecutor, AgentExecutorEvaluator } from '../../lib/langchain/executors/types';
+import { ActionsClientLlm } from '../../lib/langchain/llm/actions_client_llm';
+import {
+  indexEvaluations,
+  setupEvaluationIndex,
+} from '../../lib/model_evaluator/output_index/utils';
+
+const AGENT_EXECUTOR_MAP: Record<string, AgentExecutor> = {
+  DefaultAgentExecutor: callAgentExecutor,
```

Comment:
consider adding a comment to point developers to this map, as it's the place where the `Agents` dropdown in the UI gets its values from, e.g. `DefaultAgentExecutor`